src/runtime/contrib/tensorrt/tensorrt_module.cc

```diff
@@ -0,0 +1,184 @@
+/* * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <stdlib.h>
+#include <tvm/node/serialization.h>
+#include <tvm/relay/expr_functor.h>
+#include <tvm/relay/type.h>
+#include <tvm/runtime/ndarray.h>
+
+#include <fstream>
+#include <string>
+#include <unordered_map>
+#include <vector>
+#include "../../file_util.h"
+#include "tensorrt_builder.h"
+
+#include "NvInfer.h"
+
+namespace tvm {
+namespace runtime {
+
+Module TensorRTModuleCreate(const std::string& serialized_subgraph);
+
+class TensorRTModule : public runtime::ModuleNode {
+ public:
+  explicit TensorRTModule(const std::string& serialized_subgraph)
+      : serialized_subgraph_(serialized_subgraph) {}
+
+  ~TensorRTModule() {
+    for (auto& it : trt_engine_cache_) {
+      it.second.context->destroy();
+      it.second.engine->destroy();
+    }
+  }
+
+  PackedFunc GetFunction(const std::string& name,
+                         const ObjectPtr<Object>& sptr_to_self) final {
+    // Generate an external packed function
+    return PackedFunc([this, name](tvm::TVMArgs args, tvm::TVMRetValue* rv) {
+      auto it = trt_engine_cache_.find(name);
```

Comment:
can trt_engine_cache_ contains more than one element? TensorRTModule is created for each subgraph right? I'm assuming TensorRT will compile the subgraph as one "TensorRT op".

Reply:
Currently there can only be one entry. But once multiple functions are supported in `EnableTrt`/partitioning and `TensorRTModuleCodegen` is updated, it will be possible for there to be more than one entry.